bindgen/ir/item.rs

```diff
@@ -922,8 +923,17 @@ impl Item {
         let name = names.join("_");
 
         let name = if opt.user_mangled == UserMangled::Yes {
+            let item_info = ItemInfo {
+                name: &name,
+                kind: match self.kind() {
+                    ItemKind::Function(..) => {
+                        crate::callbacks::ItemKind::Function
+                    }
+                    _ => crate::callbacks::ItemKind::Var,
+                },
```

Comment:
Is this correct? There are no tests for this but reporting everything that isn't a function a `Var` seems totally wrong.

Reply:
Good catch, it isn't really correct. It looks like the ir ItemKind type has 4 variants, while the callback ItemKind type has only 2. Modules and Types are the variants missing, I'll add those to the callback enum so that they can be correctly handled.